### PR TITLE
Add Github Actions CI for charm operators

### PR DESCRIPTION
**Pull request by knkski**
**What this PR does / why we need it**:

Adds Github Actions CI for charm operators, to ensure that there's no breakages / regressions

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

Adds new bundle for CI purposes that points at the `latest` tags instead of particular revisions.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/test-microk8s.yaml
Comment (andreyvelich) [inline .github/workflows/test-microk8s.yaml:36]: Do you need to update other images also?
For example: `docker.io/kubeflowkatib/file-metrics-collector` ?

File: .github/workflows/test-microk8s.yaml
Comment (andreyvelich) [inline .github/workflows/test-microk8s.yaml:1]: Should you name this file `test-microk8s.yml` ?
I don't see that this action is running on this PR.

**Review by andreyvelich — COMMENTED** [review]
Thank you for this contribution @knkski!
I left comments.

File: .github/workflows/test-microk8s.yaml
Comment (andreyvelich) [inline .github/workflows/test-microk8s.yaml:19]: Since you are not using any pre-existing GitHub actions (e.g. `uses: actions/checkout@v2`) to install microk8s, does GitHub actions have any advantages  compare to Travis ?
What do you think @gaocegege @johnugeorge @knkski ?

Comment (gaocegege) [reply .github/workflows/test-microk8s.yaml:19]: https://github.com/helm/kind-action We can use this action, I think.

File: .github/workflows/test-microk8s.yaml
Comment (knkski) [reply .github/workflows/test-microk8s.yaml]: I actually took out the usage of `./scripts/v1beta/build.sh`, as building all of those images took up too much disk space.

I was thinking about only building the images as they get included in testing, and right now the only test is a simply connectivity test. I'm going to be working on expanding that, but wanted to keep it simple initially.

File: .github/workflows/test-microk8s.yaml
Comment (knkski) [reply .github/workflows/test-microk8s.yaml]: Updated the name of the file to match this name. As far as why it's not showing up, do you have to enable actions in settings or anything? I'm not sure exactly why it's not showing up.

File: .github/workflows/test-microk8s.yaml
Comment (knkski) [reply .github/workflows/test-microk8s.yaml]: Updated it to use the `balchua/microk8s-actions@v0.2.1` action. I have a slight preference for github actions due to the UI being better about scrolling through large amounts of output, but can switch to Travis CI if desired.

File: .github/workflows/test-microk8s.yaml
Comment (andreyvelich) [reply .github/workflows/test-microk8s.yaml]: @Bobgy Can you help us with enabling GitHub actions on the Katib repository, please ?
I think it should be pretty straightforward: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository#managing-github-actions-permissions-for-your-repository
/cc @gaocegege @johnugeorge

**Comment by gaocegege** [comment]
ping @andreyvelich 

/lgtm

**Comment by andreyvelich** [comment]
/hold
Hold for the GitHub actions activation.
Check this: https://github.com/kubeflow/katib/pull/1407#discussion_r537919153.

**Comment by knkski** [comment]
Pushed up an update that bumped the version of `balchua/microk8s-actions` from `v0.2.1` to `v0.2.2`

**Comment by Bobgy** [comment]
https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository

The page doesn't load atm, I'll take a look later.
A constant reminder, whenever you are blocked for configuring the repo, you should consider moving to your own org.

**Comment by Bobgy** [comment]
I don't think I need to change anything.
> By default, GitHub Actions is enabled on all repositories and organizations.

so is katib repo

**Comment by andreyvelich** [comment]
> I don't think I need to change anything.
> 
> > By default, GitHub Actions is enabled on all repositories and organizations.
> 
> so is katib repo

Thank you for helping with this!
@Bobgy So In the Katib settings in the Action permissions you are able to see that we allow all actions ?
Like this: 
Screenshot 2020-12-10 at 18 56 02

**Comment by andreyvelich** [comment]
@knkski In the meantime, can you try to create PR on your local `master` branch from you branch on the forked Katib repository ?
Then, we can check if actions will be triggered correctly.
Like I did here: https://github.com/andreyvelich/katib/pull/3.

**Comment by knkski** [comment]
Created https://github.com/knkski/katib/pull/2 to test in the meantime

**Comment by andreyvelich** [comment]
@knkski Thanks!
Btw, @yuzisun how did you enable GitHub actions for the KFServing ?
Have you changed any repo settings before creating the workflow ?

**Comment by andreyvelich** [comment]
Thanks @yuzisun, it's strange why KFServing and Katib have various GitHub actions policies.

**Comment by Bobgy** [comment]
Yes, it shows all actions are enabled

**Comment by andreyvelich** [comment]
I believe GitHub actions should be activated once we merge this PR. 
Check this: https://github.com/kubeflow/tf-operator/pull/1196#pullrequestreview-556771919.
/hold cancel
/retest

**Review by andreyvelich — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
